src/modules/communication/utils/Gcode.h

```diff
@@ -48,9 +48,10 @@ class Gcode {
 
         StreamOutput* stream;
         string txt_after_ok;
+		
+        char *command;
 
```

Comment:
I am not comfortable with this. it is private for a reason.

Reply:
Jim:

There are several (many?) cases where access to the gcode line as a string makes total sense, and is being handled in a non-standard, inconsistent, non-modular way right in GcodeDispatch (see my further comments to this later in the second change request below).  If you are concerned about some other module **_changing_** the string, or being able to, I can understand that.  However, it could be done with an object access and iterator function.....but do you relaly think that is truly necessary?  That would mean that Smoothieware would have to incorporate unchecked, uncooperative, or malicious code to make hiding `*command` a necessity.  Right?  Interested in your further thoughts on this...

Best,
Eric